litert/vendors/intel_openvino/dispatch/device_context.cc

```diff
@@ -0,0 +1,185 @@
+                                  "MMAP failed for tensor buffer");
+
+      auto context = core_.get_default_context("NPU")
+                         .as<ov::intel_npu::level_zero::ZeroContext>();
+      std::vector<int32_t> ov_shape_vec(tensor_type.layout.rank);
+      for (int i = 0; i < ov_shape_vec.size(); i++)
+        ov_shape_vec[i] = tensor_type.layout.dimensions[i];
+
+      // TODO: change f32 to ov_element_type fetched from TensorType
+      auto remote_tensor = context.create_tensor(
+          ov::element::f32, ov::Shape{ov_shape_vec.begin(), ov_shape_vec.end()},
+          buffer_fd);
+      tensor_handle_map_.emplace((LiteRtTensorBufferHandle)mmap_handle,
+                                 remote_tensor);
+      return (uint64_t)mmap_handle;
+
+#else
+      return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                "DmaBuf support is missing on this platform");
+#endif  // LRT_HAS_DMABUF_SUPPORT
+      break;
+    }
+    case kLiteRtTensorBufferTypeAhwb: {
+#if LITERT_HAS_AHWB_SUPPORT
+      AHardwareBuffer *ahwb;
+      LITERT_RETURN_IF_ERROR(
+          LiteRtGetTensorBufferAhwb(tensor_buffer, &ahwb),
+          litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                             "Failed to get LiteRT Tensor Buffer for AHWB"));
+
+      auto fd_exp = GetFdFromUnixHandle(ahwb);
+      int fd = fd_exp.Value();
+      LITERT_RETURN_IF_ERROR(
+          fd != -1, litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                       "Failed to get FD from unix handle"));
+
+      std::vector<int32_t> ov_shape_vec(tensor_type.layout.rank);
+      for (int i = 0; i < ov_shape_vec.size(); i++)
+        ov_shape_vec[i] = tensor_type.layout.dimensions[i];
+      auto context = core_.get_default_context("NPU")
+                         .as<ov::intel_npu::level_zero::ZeroContext>();
+      ov::RemoteTensor tensor = context.create_tensor(
+          ov::element::f32, ov::Shape{ov_shape_vec.begin(), ov_shape_vec.end()},
+          fd);
+      static int mmap_handle = 1;
+      tensor_handle_map_.emplace((LiteRtTensorBufferHandle)mmap_handle, tensor);
+      return (uint64_t)mmap_handle++;
+
+#else
+      return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                "AHWB support is missing on this platform");
+#endif  // LITERT_HAS_AHWB_SUPPORT
+      break;
+    }
+
+    default:
+      return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                "Unsupported tensor buffer type");
+  }
+}
+
+litert::Expected<void> LiteRtDispatchDeviceContextT::UnregisterTensorBuffer(
+    LiteRtTensorBufferHandle tensor_buffer_handle) {
+  auto it = tensor_handle_map_.find(tensor_buffer_handle);
+  if (it != tensor_handle_map_.end()) {
+    tensor_handle_map_.erase(tensor_buffer_handle);
+  } else
+    return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
```

Comment:
nit: It would be best to follow the Google coding guidelines, to ensure the repo is consistent. 

Here, we'd recommend `{}` for the `else` even though it fits on one line, see https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching Ideally we stick to most of the guide.